### ISSUE TITLE
Proposal: use Publications instead of Catalogues for Secondary literature

### ISSUE THREAD

**Issue by fjorba** [rism-ch/muscat#975]
Naming things is not easy, and sometimes the name depends on how things evolve.  In Muscat there is a Marc record type that sometimes is called Catalogues and sometimes Secondary literature.  Due to this ambiguity, at the beginning I didn't pay attention to it, and it took me a while to understand its real use.

Let me propose to call it Research, that groups both meanings in a single word.  This name change would also clarify and facilitate the use of this record type for using Muscat as research repository software by itself, without using and thus, without any conflict with the Sources record type.

Removing the word Catalogue from Muscat record types would also solve the confusion that sometimes arises with the /catalog route, as inherited from Blacklight.

Issue #958 exposes some background for my request.

I know that it is quite a job to change it, but there is documentation about how to do it in Rails.  However, before doing anything, I'd like to hear your thoughts about my proposal.

**Comment by xhero** [comment]
Thanks for the suggestion! This is actually a know issue, but the naming is relatively a sensitive issue and for the moment we gave it low priority as we need more discussion internally on this topic. Maybe @jenniferward can chime in with more info?

**Comment by jenniferward** [comment]
For our RISM users, I only use "Secondary literature" for this section. I can't use the term "catalogues" here with RISM users because the word has too many meanings (database, OPAC, book catalog, etc.). I've treated "catalogues" as an internal Muscat name, and I'll leave it up to the developers whether they want to change it or not. But in RISM, this section is for secondary literature ("research" is too general for RISM purposes).

**Comment by fjorba** [comment]
I understand.  I'm trying to avoid Catalogues for the reasons cited above. And the name of the record type appears on the (public) url, I thing that a better name would be a good thing.

What about References?

**Comment by fjorba** [comment]
I think, @jenniferward, that your opinion matters; probably I can convince more easily @xhero if you could please could express your preference as an alternative to Catalogues. I understand that, the way Rails applications are build, a single word is a must, so Secondary Literature is hard to use.

I have been browsing the (paper version!) of the Roget's College Thesaurus (1978 edition!) that I have at home, and the only satisfactory alternatives I've found are References and Publications, but you may think of others.

Thanks.

**Comment by jenniferward** [comment]
"References" works for me, as does "publications," maybe also "citations". (My Roget is from 1998, lol)
The URLs do get tricky when sharing with other users, as the word "catalogues" doesn't make the purpose immediately clear. But this is a minor nuisance for my purposes.

**Comment by fjorba** [comment]
Thanks, @jenniferward.  Since I wrote my alternatives, I have been thinking about them in the context of institutional and research repositories.  I'd like to comment it out with my colleagues here, but I'd say that the word that has less ambiguity (in the sense that is less likely to be confused with anything else) would be Publications.

Citation can be easily understood as citation style (MLA, Chicago, etc.), and a Blacklight library catalog like Princeton uses it (ex: https://catalog.princeton.edu/catalog/6243820, Cite link).

References is much used as bibliographic references at the end of the work, that sometimes are also shown in the bibliographic reference proper (ex: https://www.scopus.com/results/citedbyresults.uri?sort=plf-f&cite=2-s2.0-84860363205&src=s&imp=t&sid=fd149194b5909165648c8019467e2c24&sot=cite&sdt=a&sl=0&origin=inward&editSaveSearch=&txGid=d3296652293f710f55dfe3177cd32a9b).

However, I cannot think of any other sense for Publication than the one we are discussing here: published works about a work, a subject or an author or composer.  So, for the time being, I'd favour Publications.

Thanks again.

**Comment by fjorba** [comment]
Ok, my librarian colleagues also accept Publications.

@xhero, I volunteer to create the patch, unless somebody else prefers to do it instead. No problem for me.  However, given that it requires a considerable effort (at least for me, still a Ruby and Rails newbie), I'd like your commitment that, if I'm able to do it technically correct, you will apply it upstream.  I have posted some pull requests that should be apply before this one.

**Comment by xhero** [comment]
Well on my side there is just really the technical part of renaming a model, which is quite major and for our guidelines would need to be done in a major release + migration of all the data. Nothing impossible.
My concern is more if everyone in the ZR is ok with this, as it is very major and there is no coming back from it. At least I will not make a release to revert it :)
Since this would be very big change in the system it is imperative that everybody in the ZR is ok with this change and it is approved before we can proceed.
The light weight alternative is that we just change the label, basically as we do now: the model is still called "catalogues" (from the time when it was just catalogues of works) but the label says "secondary literature". This would be a very easy change, but I still need approval from ZR on it.

**Comment by HirschSt** [comment]
@xhero thanks a lot. Also keep in mind that changing the model name concerns interfaces too, eg SRU with https://muscat.rism.info/sru/catalogues?operation=searchRetrieve&version=1.1&query=author=Eitner&maximumRecords=20&startRecord=5 for example. Changing only the label would be a very good compromise (although "catalogue" and "catalog" (blacklight) is driving me crazy some times...)

**Comment by fjorba** [comment]
I understand, and we can wait. However, I really would like to insist in this name change.

Muscat has many virtues, and can become only major institutional repository with 100% native Marc21 support. I have concluded that both communities (to simplify: the musicologists and the institutional and research institutions) can both benefit of this with a larger workforce.  The flexibility that Muscat Marc configuration files, and the existence of those two bibliographic records (sources and secondary literature) allows that each has its own ecosystem, with authority records serving both.  The larger Blacklight community is a wonderful example of inter-institutional collaboration.

But we do need a better name. Seeing /admin/catalogues/new is discouraging; /admin/publications/new would be great.

We can wait until the consensus is reached. Thanks.

**Comment by fjorba** [comment]
By the way, this migration would be a great moment to convert `marc_source` to `mediumtext`, as suggested in #792.  Curiously, the catalogues table was the only one not suggested in your examples; at that time, I still wasn't aware of its existence, and it was because if this confusing name.

**Comment by xhero** [comment]
I personally don't love the "catalogues" name as it does not really describe what is in there, so I'm not completely opposed to changing it, we need a some more internal discussion on this.

**Comment by HirschSt** [comment]
Voting of ZR tends to -> Publications

**Comment by fjorba** [comment]
On 21 september, you @xhero wrote:
> Well on my side there is just really the technical part of renaming a model, which is quite major and for our guidelines would need to be done in a major release + migration of all the data. Nothing impossible.

Just to make I sure that we all understand the same: does this sentence mean that you are the one to create the patch (migration, rename variables and the like) for 7.0?

**Comment by xhero** [comment]
yes! wish me luck! it is a bit verbose but not impossible

**Comment by fjorba** [comment]
Buona fortuna!  And please update this task with your progress, and plase give me some days after doing it so I can advance with #1013, as it depends on your task.  I'd like to have more than one template for 7.0, so the others will already have the db schema migrated.  Otherwise, we'll have to wait another round.

**Comment by xhero** [comment]
Merged in develop!

**Comment by fjorba** [comment]
Sorry, I'm preparing a full set of changes for renaming the table and fields.  I didn't have time to do it before, but I'll try to finish it between today and tomorrow.
